site/content/en/docs/Integration Patterns/high-density-gameservers.md

```diff
@@ -13,34 +13,104 @@ use of resources to run multiple concurrent game sessions from within a single `
 The tradeoff here is that this requires more management on behalf of the integrated game server process and external 
 systems, since it works around the common Kubernetes and/or Agones container lifecycle.
 
-Utilising the new allocation `gameServerState` filter as well as the existing ability to edit the 
+Here are two different approaches to solving this problem with Agones:
+
+## Session/Room Counters
+
+{{< beta title="Counters And Lists" gate="CountsAndLists" >}}
+
+Utilising the allocation `gameServerState` filter as well as the new ability to add Counter capacity and counts to
+`GameServer` records at both [allocation time]({{% ref "/docs/Reference/gameserverallocation.md" %}}), and from
+within the game server process, [via the SDK][sdk-counter], means Agones is able to atomically track how many sessions 
+are available on a given a `GameServer` from the list of potentially Ready or Allocated `GameServers` when making an
+allocation request.
+
+By also using Counters, we can provide Agones the allocation metadata it needs to pack appropriately across the high 
+density `GameServer` instances as well.
+
+<a href="../../../diagrams/high-density-counters.puml.png" target="_blank">
+<img src="../../../diagrams/high-density-counters.puml.png" alt="High Density Allocation Diagram (Session Counters)" />
+</a>
+
+### Example `GameServerAllocation`
+
+The below `Allocation` will first attempt to find a `GameServer` from the `Fleet` `simple-game-server` that is already
+Allocated and also available capacity under the `rooms` Counter.
+
+If an Allocated `GameServer` does not exist with available capacity, then use the next selector to allocate a Ready
+`GameServer` from the `simple-game-server` `Fleet`.
+
+Whichever condition is met, once allocation is made against a `GameServer`, the `rooms` Counter will be incremented by
+one, thereby decrementing the available capacity of the `room` Counter on the `GameServer` instance. Once there is no 
+available capacity on the most full `GameServer`, the allocation will look for the next least full `GameServer` to
+ensure packing across `GameServer` instances.
+
+It will then be up to the game server process to decrement the `rooms` Counter via the SDK when a session comes to end,
+to increase the amount of available capacity within the `GameServer` instance.
+
+```yaml
+apiVersion: allocation.agones.dev/v1
+kind: GameServerAllocation
+spec:
+  scheduling: Packed
```

Comment:
A side note that when scheduling is packed the `priorities` are a tie-breaker after all our other logic around node utilization etc., and the `distributed` uses `priorities` as the sorting strategy that's probably worth mentioning somewhere in this doc?

Reply:
oh that's a good point! Lemme put something in there. Also remembering we have #3652 as well to update those docs. 

I've softened the language, since it's not a hard and fast rule, like you said, also put a Info box at the end.